src/mca/oob/tcp/oob_tcp_connection.c

```diff
@@ -433,7 +433,7 @@ static int tcp_peer_send_connect_ack(prrte_oob_tcp_peer_t* peer)
     offset += sizeof(hdr);
     memcpy(msg + offset, &ack_flag, sizeof(ack_flag));
     offset += sizeof(ack_flag);
-    memcpy(msg + offset, prrte_version_string, strlen(prrte_version_string));
+    memcpy(msg + offset, prrte_version_string, strlen(prrte_version_string) + 1);
```

Comment:
No issue with this, but it technically isn't required. We already zero'd the msg array and in the next line we step over the `strlen(prrte_version_string) + 1` spot, thereby leaving it as the NULL terminator for the string.

Reply:
The issue valgrind had with it is the buffer `msg` was malloc'd, not calloc'd. And since this wasn't copying in the null terminator and skipping over it, that one byte could technically be anything.

Reply:
Look at line 429 - we NULL the entire buffer. Your compiler sounds like it has a bug 😄

Reply:
or rather, I should have said valgrind has the bug - we constantly run into these bugs with them, which is why we have such a big valgrind "ignore" file

Reply:
Ah - yeah it looks like that is the case. It's probably a valgrind bug then.